internal/transport/http2_client.go

```diff
@@ -1318,7 +1318,8 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 				t.closeStream(s, se.Err(), true, http2.ErrCodeProtocol, se, nil, endStream)
 				return
 			}
-			rawStatusCode = codes.Code(uint32(code))
+			rsc := codes.Code(uint32(code))
```

Comment:
Shouldn't `ParseInt` above be `ParseUint`?

And then we need a bounds check on this as well.

Reply:
I adjusted this to `ParseUint`. What kind of bounds check should I do?

Reply:
Hmm, nevermind the bounds checks.  If we were to add a new status code, I believe we'd want to allow it to be propagated by a gRPC proxy without requiring recompilation of the proxy.  So any unsigned value is fine here.

Reply:
OK great. I think this is ready for re-review then!